File: log/log_plotter.m
clc;
clear;
close all;




T = readtable("log/pid_20260517_171059.csv");
% Convert the data from radians to degrees
T.imu_roll = T.imu_roll * 180 / pi;
T.pid_roll = T.pid_roll * 180 / pi;

T.imu_pitch = T.imu_pitch * 180 / pi;
T.pid_pitch = T.pid_pitch * 180 / pi;

subplot(2, 1, 1)
plot(T.t, T.imu_roll, T.t, T.pid_roll)
ylim([-10 10])
legend("IMU", "PID")
xlabel("Time (s)")
ylabel("Angle (deg)")
title("Roll Angle")
grid on

subplot(2, 1, 2)
plot(T.t, T.imu_pitch, T.t, T.pid_pitch)
ylim([-10 10])
legend("IMU", "PID")
xlabel("Time (s)")
ylabel("Angle (deg)")
title("Pitch Angle")
grid on

File: log/log_plotter.py
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np



def plot_log(file_name):
    fig, axes = plt.subplots(2, 1, figsize=(10, 8))
    df = pd.read_csv(file_name)
    df["imu_roll"]  = df["imu_roll"]  * 180 / np.pi
    df["imu_pitch"] = df["imu_pitch"] * 180 / np.pi
    df.plot(x="t", y=["imu_roll"],  ylim=[-20, 20], ylabel="Roll (deg)",  ax=axes[0], grid=True)
    df.plot(x="t", y=["imu_pitch"], ylim=[-20, 20], ylabel="Pitch (deg)", ax=axes[1], grid=True)
    plt.tight_layout()
    plt.savefig(file_name.replace(".csv", ".png"))


def plot_log_dh(file_name):
    fig, axes = plt.subplots(2, 1, figsize=(10, 8))
    df = pd.read_csv(file_name)
    df.plot(x="t", y=["dh_roll",  "corr_roll"],  ylabel="Roll DH (mm)",  ax=axes[0], grid=True)
    df.plot(x="t", y=["dh_pitch", "corr_pitch"], ylabel="Pitch DH (mm)", ax=axes[1], grid=True)
    plt.tight_layout()
    plt.savefig(file_name.replace(".csv", ".png"))
